javascript/ql/src/semmle/javascript/Classes.qll

```diff
@@ -210,6 +217,23 @@ class ClassDeclStmt extends @classdeclstmt, ClassDefinition, Stmt {
  * A class expression.
  */
 class ClassExpr extends @classexpr, ClassDefinition, Expr {
+  override string getName() {
+    result = ClassDefinition.super.getName()
+    or
+    exists(VarDef vd | this = vd.getSource() | result = vd.getTarget().(VarRef).getName())
+    or
+    exists(Property p |
+      this = p.getInit() and
+      result = p.getName()
+    )
+    or
+    exists(AssignExpr assign, DotExpr prop |
```

Comment:
Can you expand a bit on why you are restricting to `DotExpr` here and below? Is it to avoid numeric indices?

Reply:
It was to avoid having names that aren't valid identifiers, for no particularly good reason.

Reply:
Fair enough. I guess cases like `o.["export"] = function() { ... }` aren't worth the bother.

Reply:
That's actually a fine point, as some names are reserved in some contexts but not others. I've seen functions named `delete` a few times. I went ahead and changed it to `PropAccess` after all.